plat/allwinner/sun50i_a64/sunxi_power.c

```diff
@@ -150,10 +228,18 @@ static void setup_axp803_rails(const void *fdt)
 	for (node = fdt_first_subnode(fdt, node);
 	     node != -FDT_ERR_NOTFOUND;
 	     node = fdt_next_subnode(fdt, node)) {
+		struct axp_regulator *reg;
 		const char *name;
 		int length;
 
 		name = fdt_get_name(fdt, node, &length);
+		for (reg = regulators; reg->dt_name; reg++) {
```

Comment:
If you pull the `fdt_handle_regulator` call outside the loop, in addition to avoiding the loop in that case, you could swap the two `if` statements for `dc1sw` below, and only call it once.